src/Medidata.ZipkinTracer.Core/Medidata.ZipkinTracer.Core.csproj

```diff
@@ -58,23 +58,34 @@
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
-    <Reference Include="Thrift, Version=0.9.1.3, Culture=neutral, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\packages\Thrift.0.9.1.3\lib\net35\Thrift.dll</HintPath>
-    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="ITraceProvider.cs" />
     <Compile Include="ITracerClient.cs" />
     <Compile Include="IZipkinConfig.cs" />
     <Compile Include="Middlewares\ZipkinMiddleware.cs" />
+    <Compile Include="Models\Annotation.cs" />
+    <Compile Include="Models\AnnotationBase.cs" />
+    <Compile Include="Models\BinaryAnnotation.cs" />
+    <Compile Include="Models\Endpoint.cs" />
+    <Compile Include="Models\References\AnnotationType.cs" />
+    <Compile Include="Models\References\ZipkinConstants.cs" />
+    <Compile Include="Models\Serialization\Json\JsonAnnotation.cs" />
```

Comment:
Yes, my idea was that the serialization classes are just wrappers around the model classes which basically represent the original model data in a serializable manner. In this pattern for example the `JsonSpan` should not hold any information (or data) by itself, only the logic to transform the original data into a serializable data.

This will give the benefit that at the moment of the serialization the data will be the exactly same as in the model class (effectively serializing the real model instead of an earlier transformed copy of the model).